sensor.py

```diff
@@ -106,3 +123,95 @@ def icon(self):
     @property
     def device_class(self):
         return DEVICE_CLASS_BATTERY
+
+
+class HSTemperature(HSSensor):
+    """Representation of a HomeSeer temperature sensor device."""
+
+    @property
+    def unit_of_measurement(self):
+        if self._units is not None:
+            if str(self._units) == 'imperial':
+                return "°F"
+            elif str(self._units) == 'metric':
+                return "°C"
+            else:
+                return "?"
+        else:
+            return '?'
+
+    @property
+    def icon(self):
+        if self._units is not None:
+            if str(self._units) == 'imperial':
+                if self.state < 50:
+                    return 'mdi:thermomter-low'
+                elif self.state > 90:
+                    return 'mdi:thermometer-high'
+                else:
+                    return 'mdi:thermometer'
+            elif str(self._units) == 'metric':
+                if self.state < 10:
+                    return 'mdi:thermomter-low'
+                elif self.state > 33:
+                    return 'mdi:thermometer-high'
+                else:
+                    return 'mdi:thermometer'
+            else:
+                return 'mdi:thermometer'
+        else:
+            return 'mdi: thermometer'
```

Comment:
Same as above - if we know what unit the data is coming from HomeSeer in, then we can figure out what the icons should be always. Knowing what unit Home Assistant is using doesn't help us here.

Reply:
See above.  Is it not "safe" to assume that the unit system set in HA is the same as what would have been set in HS?